src/helpers/text_helper.ts

```diff
@@ -224,6 +228,90 @@ export function getContextFontSize(font: string): Pixel {
   return Number(font.match(pxRegex)?.[1]);
 }
 
+export function fittingString(ctx, str: string, maxWidth: number) {
+  var width = ctx.measureText(str).width;
+  var ellipsis = "…";
+  var ellipsisWidth = ctx.measureText(ellipsis).width;
```

Comment:
measure text can be expensive. Can `computeTextWidth` be used here (it's cached)